---
title: "plot figures"
author: "Chunmei Li"
date: "2018/10/17"
output: html_document
editor_options: 
  chunk_output_type: console
---
# History

# To Do
* Label on saved file or maker a comment at "ggsave" line with fig or table numbers to become reader friendly scripts (KN, 102318)

```{r}
library(tidyverse)
library(reshape2)
library(edgeR)
```
#load dge data
```{r}
if(Sys.info()["user"]=="nozue") { 
list.files(file.path("~","Box","Chunmei_Myc_Paper","manuscripts","input"))
load(file.path("~","Box","Chunmei_Myc_Paper","manuscripts","output","dge.Rdata")) # using Box folder! 
}
if(Sys.info()["user"]=="LCM") {
load(file.path("C","Users","LCM","Box","manuscripts","output","dge.Rdata")) # using Box folder! 
} # Does this work?
dge.data$samples
head(cpm(dge.data))
log2_cpm <- log2(cpm(dge.data)+1)
head(log2_cpm)
rownames(log2_cpm) <- gsub("\\.\\d", "", rownames(log2_cpm))

#subset Col and myc234 data

log2_cpm <- log2_cpm[, 1:32]
head(log2_cpm)
```
# finding DEG files in csv
```{r}
#only retain the Col and myc234 data
if(Sys.info()["user"]=="nozue") { 
DEG.objs2<-list.files(file.path("~","Box","Chunmei_Myc_Paper","manuscripts","output","figures_and_tables"), pattern = ".rH\\.csv")[1:6]
load(file.path("~","Box","Chunmei_Myc_Paper","manuscripts","output","dge.Rdata")) # using Box folder! 
}
if(Sys.info()["user"]=="LCM") {
DEG.objs2<-list.files(file.path("C","Users","LCM","Box","manuscripts","output","figures_and_tables"),pattern = ".rH\\.csv")[1:6] # using Box folder! 
} # Does this work?
DEG.objs2
```

# reading all csv files
```{r}
# read csv file into a large list
if(Sys.info()["user"]=="nozue") { 
DEG.list<-lapply(DEG.objs2, function(x) read_csv(paste(file.path("~","Box","Chunmei_Myc_Paper","manuscripts","output","figures_and_tables"),"/",x,sep="")))
names(DEG.list)<-gsub(".csv","",DEG.objs2)
}
if(Sys.info()["user"]=="LCM") { 
DEG.list<-lapply(DEG.objs2, function(x) read_csv(paste(file.path("C","Users","LCM","Box","manuscripts","output","figures_and_tables"),"/",x,sep="")))
names(DEG.list)<-gsub(".csv","",DEG.objs2)
}

```

# expression pattern of genes differentially expressed in Col after 1h treatment compared to high red/far-red ratio;

```{r DEgenes expression pattern}
#tp=1 or 49, 
for (i in 1:6) {
  genes.up <- DEG.list[[i]] %>% filter(DEG.list[[i]][3] >0 )
genes.down <- DEG.list[[i]] %>% filter(DEG.list[[i]][3]<0)

if (str_detect(names(DEG.list)[i],"1h")) {
 #only leave the gene id and sample_id colum
plot.data.up <- merge(log2_cpm, genes.up, by.x="row.names", by.y="gene_id") 
#head(plot.data.up)
plot.data.up <- plot.data.up %>% 
  dplyr::select(1, colnames(plot.data.up)[-1][grep("1h",colnames(log2_cpm))], "name") %>% 
  unite(row.names,c("Row.names","name")) %>% 
  gather(sample,value,-1) %>% separate(sample,c("genotype","time_hr","trt","rep"),sep="_") 
plot.up <- plot.data.up %>% ggplot(aes(x=genotype,y=value,color=trt))  + geom_boxplot() + facet_wrap(~row.names,scale="free",labeller = label_wrap_gen(width=30), ncol = 4)+theme(strip.text = element_text(size=5))+ggtitle(paste(names(DEG.list)[i], "up", sep = "."))+ ylab("log2(cpm)")
plot.up
if(Sys.info()["user"]=="nozue") {
ggsave(plot.up, file = file.path ("~","Box","Chunmei_Myc_Paper","manuscripts","output","figures_and_tables", paste(names(DEG.list)[i], "up.pdf", sep = ".")), width = 10,height = 10, units = "in", limitsize = FALSE)
}
if(Sys.info()["user"]=="LCM") {
ggsave(plot.up, file = file.path ("C","Users","LCM","Box","manuscripts","output","figures_and_tables", paste(names(DEG.list)[i], "up.pdf", sep = ".")), width = 10,height = 10, units = "in", limitsize = FALSE)
}

plot.data.down <- merge(log2_cpm, genes.down, by.x="row.names", by.y="gene_id") 
#head(plot.data.down)
plot.data.down <- plot.data.down %>% dplyr::select(1, colnames(plot.data.down)[-1][grep("1h",colnames(log2_cpm))], "name") %>% unite(row.names,c("Row.names","name")) %>% gather(sample,value,-1) %>% separate(sample,c("genotype","time_hr","trt","rep"),sep="_") 
plot.down <- plot.data.down %>% ggplot(aes(x=genotype,y=value,color=trt))  + geom_boxplot() + facet_wrap(~row.names,scale="free",labeller = label_wrap_gen(width=30), ncol = 4)+theme(strip.text = element_text(size=5))+ggtitle(paste(names(DEG.list)[i], "down", sep = "."))+ ylab("log2(cpm)")
plot.down
if(Sys.info()["user"]=="nozue") {
ggsave(plot.down, file = file.path ("~","Box","Chunmei_Myc_Paper","manuscripts","output","figures_and_tables", paste(names(DEG.list)[i], "down.pdf", sep = ".")), width = 10,height = 10, units = "in",limitsize = FALSE) }
if(Sys.info()["user"]=="LCM") {
ggsave(plot.down, file = file.path ("C","Users","LCM","Box","manuscripts","output","figures_and_tables", paste(names(DEG.list)[i], "down.pdf", sep = ".")), width = 10,height = 10, units = "in",limitsize = FALSE) }
} else if (str_detect(names(DEG.list)[i],"49h")) {
 #only leave the gene id and sample_id colum
plot.data.up <- merge(log2_cpm, genes.up, by.x="row.names", by.y="gene_id") 
#head(plot.data.up)
plot.data.up <- plot.data.up %>% dplyr::select(1, colnames(plot.data.up)[-1][grep("49h",colnames(log2_cpm))], "name") %>% unite(row.names,c("Row.names","name")) %>% gather(sample,value,-1) %>% separate(sample,c("genotype","time_hr","trt","rep"),sep="_") 
plot.up <- plot.data.up %>% ggplot(aes(x=genotype,y=value,color=trt))  + geom_boxplot() + facet_wrap(~row.names,scale="free",labeller = label_wrap_gen(width=30), ncol = 4)+theme(strip.text = element_text(size=5))+ggtitle(paste(names(DEG.list)[i], "up", sep = "."))+ ylab("log2(cpm)")
plot.up
if(Sys.info()["user"]=="nozue") {
ggsave(plot.up, file = file.path ("~","Box","Chunmei_Myc_Paper","manuscripts","output","figures_and_tables", paste(names(DEG.list)[i], "up.pdf", sep = ".")), width = 10,height = 10, units = "in",limitsize = FALSE)}
if(Sys.info()["user"]=="LCM") {
ggsave(plot.up, file = file.path ("C","Users","LCM","Box","manuscripts","output","figures_and_tables", paste(names(DEG.list)[i], "up.pdf", sep = ".")), width = 10,height = 10, units = "in",limitsize = FALSE)}

plot.data.down <- merge(log2_cpm, genes.down, by.x="row.names", by.y="gene_id") 
#head(plot.data.down)
plot.data.down <- plot.data.down %>% dplyr::select(1, colnames(plot.data.down)[-1][grep("49h",colnames(log2_cpm))], "name") %>% unite(row.names,c("Row.names","name")) %>% gather(sample,value,-1) %>% separate(sample,c("genotype","time_hr","trt","rep"),sep="_") 
plot.down <- plot.data.down %>% ggplot(aes(x=genotype,y=value,color=trt))  + geom_boxplot() + facet_wrap(~row.names,scale="free",labeller = label_wrap_gen(width=30), ncol = 4)+theme(strip.text = element_text(size=5))+ggtitle(paste(names(DEG.list)[i], "down", sep = "."))+ ylab("log2(cpm)")
plot.down

if(Sys.info()["user"]=="nozue") {
ggsave(plot.down, file = file.path ("~","Box","Chunmei_Myc_Paper","manuscripts","output","figures_and_tables", paste(names(DEG.list)[i], "down.pdf", sep = ".")), width = 10,height =10, units = "in",limitsize = FALSE) }
if(Sys.info()["user"]=="LCM") {
ggsave(plot.down, file = file.path ("C","Users","LCM","Box","manuscripts","output","figures_and_tables", paste(names(DEG.list)[i], "down.pdf", sep = ".")), width = 10,height =10, units = "in",limitsize = FALSE) }

}

}
```


#plot genes in each enriched GO terms
##import data
```{r eval=FALSE}
genes.in.GO.terms <- read.csv("../output/genes.in.enriched.BP.category.csv", header = T)

head(genes.in.GO.terms)
```

```{r eval=FALSE}
levels(genes.in.GO.terms$sample)
```

```{r eval=FALSE}
data.plot.all <- merge(log2_cpm, genes.in.GO.terms, by.x="row.names", by.y="AGI")
for (i in levels(genes.in.GO.terms$sample)) {
  
  if (str_detect(i,"1h")) {
    data.plot <- data.plot.all %>% filter(sample==i) %>% unite(AGI_name, "Row.names", "name") %>% unite(GO_terms, "category", "term") %>% dplyr::select(colnames(data.plot.all)[-1][grep("1h",colnames(log2_cpm))], "AGI_name", "GO_terms") %>% gather(sample,value,-(17:18)) %>% separate(sample,c("genotype","time_hr","trt","rep"),sep="_")
  plot <-data.plot %>% ggplot(aes(x=genotype,y=value,color=trt))  + geom_boxplot() + facet_wrap(GO_terms~AGI_name,scale="free",labeller = label_wrap_gen(width=30), ncol = 4)+theme(strip.text = element_text(size=5))+ggtitle(paste("genes_in_each_enriched_GOterms_in", i, sep = "_"))+ ylab("log2(cpm)")
 
   
  } else if (str_detect(i,"49h")) {
    
    data.plot <- data.plot.all %>% filter(sample==i) %>% unite(AGI_name, "Row.names", "name") %>% unite(GO_terms, "category", "term") %>% dplyr::select(colnames(data.plot.all)[-1][grep("49h",colnames(log2_cpm))], "AGI_name", "GO_terms") %>% gather(sample,value,-(17:18)) %>% separate(sample,c("genotype","time_hr","trt","rep"),sep="_")
    plot <-data.plot %>% ggplot(aes(x=genotype,y=value,color=trt))  + geom_boxplot() + facet_wrap(GO_terms~AGI_name,scale="free",labeller = label_wrap_gen(width=30), ncol = 4)+theme(strip.text = element_text(size=5))+ggtitle(paste("genes_in_each_enriched_GOterms_in", i, sep = "_"))+ ylab("log2(cpm)")

  }
   
  ggsave(plot, file = file.path ("../figures", paste("genes.in.each.enriched.GOterms.in", i,"pdf", sep = ".")), width = 10,height =100, units = "in",limitsize = FALSE) 
}
```



#test

```{r eval=FALSE}
if (str_detect(i,"1h")) {
    data.plot <- data.plot.all %>% filter(sample=="Col_1h_rH_up") %>% unite(AGI_name, "Row.names", "name") %>% unite(GO_terms, "category", "term") %>% dplyr::select(colnames(data.plot.all)[-1][grep("1h",colnames(log2_cpm))], "AGI_name", "GO_terms") %>% gather(sample,value,-(17:18)) %>% separate(sample,c("genotype","time_hr","trt","rep"),sep="_")
  plot <-data.plot %>% ggplot(aes(x=genotype,y=value,color=trt))  + geom_boxplot() + facet_wrap(GO_terms~AGI_name,scale="free",labeller = label_wrap_gen(width=30), ncol = 4)+theme(strip.text = element_text(size=10))+ggtitle(paste("genes_in_each_enriched_GOterms_in", "Col_1h_rH_up", sep = "_"))+ ylab("log2(cpm)")
 
   
  }
```

#plot genes in each enriched hormone responsive category
##import data
```{r eval=FALSE}
genes.in.hormone.category <- read.csv("../output/genes.in.enriched.hormone.responsive.category.csv", header = T)

head(genes.in.hormone.category)
```

```{r eval=FALSE}
levels(genes.in.hormone.category$sample)
```

```{r eval=FALSE}
data.plot.all <- merge(log2_cpm, genes.in.hormone.category, by.x="row.names", by.y="AGI")
for (i in levels(genes.in.hormone.category$sample)) {
  
  if (str_detect(i,"1h")) {
    data.plot <- data.plot.all %>% filter(sample==i) %>% unite(AGI_name, "Row.names", "name") %>% dplyr::select(colnames(data.plot.all)[-1][grep("1h",colnames(log2_cpm))], "AGI_name", "category") %>% gather(sample,value,-(17:18)) %>% separate(sample,c("genotype","time_hr","trt","rep"),sep="_")
  plot <-data.plot %>% ggplot(aes(x=genotype,y=value,color=trt))  + geom_boxplot() + facet_wrap(category~AGI_name,scale="free",labeller = label_wrap_gen(width=30), ncol = 4)+theme(strip.text = element_text(size=5))+ggtitle(paste("genes_in_each_enriched_hormone_responsive_category_in", i, sep = "_"))+ ylab("log2(cpm)")
 
   
  } else if (str_detect(i,"49h")) {
    
    data.plot <- data.plot.all %>% filter(sample==i) %>% unite(AGI_name, "Row.names", "name") %>% dplyr::select(colnames(data.plot.all)[-1][grep("49h",colnames(log2_cpm))], "AGI_name", "category") %>% gather(sample,value,-(17:18)) %>% separate(sample,c("genotype","time_hr","trt","rep"),sep="_")
    plot <-data.plot %>% ggplot(aes(x=genotype,y=value,color=trt))  + geom_boxplot() + facet_wrap(category~AGI_name,scale="free",labeller = label_wrap_gen(width=30), ncol = 4)+theme(strip.text = element_text(size=5))+ggtitle(paste("genes_in_each_enriched_hormone_responsive_category_in", i, sep = "_"))+ ylab("log2(cpm)")

  }
   
  ggsave(plot, file = file.path ("../figures", paste("genes.in.each.enriched.hormone.responsive.category.in", i,"pdf", sep = ".")), width = 10,height =80, units = "in",limitsize = FALSE) 
}
```
