---
title: "Untitled"
output: html_document
---

```{r}
library(ShortRead);library(goseq);library(GO.db);library("annotate")#use annotate to add terms for each GO category
```

```{r}
library(limma)
library(tidyverse)
library(ggforce)
```


```{r}
getwd()
At_cdna<-Biostrings::readDNAStringSet("../reference/TAIR10_cdna_20110103_representative_gene_model_updated.1")
At_cdna

```


```{r}
Atgoslim.TAIR<-read_tsv("../reference/ATH_GO_GOSLIM.txt",skip=2,col_names = FALSE )
```
```{r}
load("../reference/At.hormone.responsive.list.Rdata")
head(At.hormone.responsive.list)
```


```{r}
head(Atgoslim.TAIR)
Atgoslim.TAIR %>% filter(X8=="P",X1=="AT4G38360") %>% dplyr::select(1,5,6,8,9,10)###what does this mean? means BP  biological process
Atgoslim.TAIR.BP <-Atgoslim.TAIR%>% filter(X8=="P")
head(Atgoslim.TAIR.BP)
Atgoslim.TAIR.BP.list<-tapply(Atgoslim.TAIR.BP$X6,Atgoslim.TAIR.BP$X1,c)
head(Atgoslim.TAIR.BP.list)
#save(Atgoslim.TAIR.BP.list, list,file=file.path("../output","Atgoslim.TAIR.BP.list.Rdata"))

Atgoslim.TAIR.CC <-Atgoslim.TAIR%>% filter(X8=="C")
head(Atgoslim.TAIR.CC)
Atgoslim.TAIR.CC.list<-tapply(Atgoslim.TAIR.CC$X6,Atgoslim.TAIR.CC$X1,c)
head(Atgoslim.TAIR.CC.list)
#save(Atgoslim.TAIR.CC.list, list,file=file.path("../output","Atgoslim.TAIR.CC.list.Rdata"))
```
#import expressed genes 

```{r}
expressed.genes <- read_csv("../output/expressed_genes_counts.csv")
expressed.genes <- expressed.genes$target_id
expressed.genes <- as.data.frame(expressed.genes)
head(expressed.genes)
colnames(expressed.genes) <- "gene_id"
expressed.genes$gene_id <- gsub("\\.\\d", "", expressed.genes$gene_id)
head(expressed.genes)
```



```{r}
GOseq.Atgoslim.BP.list.ORA<-function(genelist,padjust=0.05,ontology="BP",custom.category.list=Atgoslim.TAIR.BP.list,cdna=At_cdna,expressed.genes=expressed.genes) { # return GO enrichment table, padjus, padjust=0.05. 
  bias<-nchar(cdna)#to get the length of each gene
  names(bias)<-tibble(AGI=names(cdna)) %>% separate(AGI,into="AGI2",pattern="|",extra="drop") %>% dplyr::select(AGI2) %>% as_vector()# to get the AGI of each gene
  table(duplicated(names(bias)))# to see if there is duplicated AGI
  #TF<-(names(bias) %in% genelist)*1
  bias <- bias[names(bias)%in%expressed.genes$gene_id] # use only expressed genes in the experiment as reference
  TF<-as.integer(names(bias) %in% genelist)
  names(TF)<-names(bias)
  #print(TF)
  pwf<-nullp(TF,bias.data=bias)
  #print(pwf$DEgenes)
  GO.pval <- goseq(pwf,gene2cat=custom.category.list,use_genes_without_cat=TRUE) # format became different in new goseq version (021111). Does not work (042716)
  #head(GO.pval) 
  if(ontology=="BP") {
    GO.pval2<-subset(GO.pval,ontology=="BP")
  } else if(ontology=="CC") {
    GO.pval2<-subset(GO.pval,ontology=="CC")
  } else {
    GO.pval2<-subset(GO.pval,ontology=="MF")
  }
  # calculating padjust by BH  
  GO.pval2$over_represented_padjust<-p.adjust(GO.pval2$over_represented_pvalue,method="BH")
  if(GO.pval2$over_represented_padjust[1]>padjust) return("no enriched GO")
  else {
    enriched.GO<-GO.pval2[GO.pval2$over_represented_padjust<padjust,] 
    print("enriched.GO is")
    print(enriched.GO)
    
    ## write Term and Definition 
    for(i in 1:dim(enriched.GO)[1]) {
      if(is.null(Term(GOTERM[enriched.GO[i,"category"]]))) {next} else {
      enriched.GO$Term[i]<-Term(GOTERM[[enriched.GO[i,"category"]]])
      enriched.GO$Definition[i]<-Definition(GOTERM[[enriched.GO[i,"category"]]])
      }
    }
    return(enriched.GO)
  }
}

```


```{r}
DEG.objs2 <- list.files(path=file.path("..","output"), pattern = ".rH\\.csv")[c(1,2,3,4,5,6)]
```


```{r}
# read csv file
DEG.list<-lapply(DEG.objs2, function(x) read_csv(paste(file.path("..","output"),"/",x,sep="")))
names(DEG.list)<-gsub(".csv","",DEG.objs2)


```

```{r}
# single coefficient
names(DEG.list)

```
#summary figure
```{r}
remove(summary.data)
summary.data <- data.frame(matrix(nrow = 8, ncol = 4))
colnames(summary.data) <- c("number_of_genes", "FC", "genotype", "time_point")
for (n in 1:4) {
summary.data$number_of_genes[n] <- nrow(DEG.list[[n]]%>% filter(DEG.list[[n]][3] > 0)) 
summary.data$FC[n] <- "up"

}

for (n in 1:4) {
summary.data$number_of_genes[n+4] <- nrow(DEG.list[[n]]%>% filter(DEG.list[[n]][3] < 0)) 
summary.data$FC[n+4] <- "down"
}
 
summary.data$FC <- as.factor(summary.data$FC)
levels(summary.data$FC)
summary.data$FC <- relevel(summary.data$FC, ref = "up")
summary.data$genotype <- c("Col", "Col", "myc234", "myc234", "Col", "Col", "myc234", "myc234")
  
summary.data$time_point <- c("1h", "49h", "1h", "49h", "1h", "49h", "1h", "49h")

 p <- ggplot(summary.data) + geom_bar(stat="identity", aes(x=FC, y=number_of_genes, fill=FC)) + facet_grid(time_point~genotype) + theme(axis.title.x=element_blank(),legend.position = "none") + ylab("number of misregulated genes")
ggsave(filename = file.path("../output/summary of misregulated genes.pdf"), p, height = 5, width = 5, units = "in")
```

# venn diagram

```{r}
vennDiagram()
```



#for double coeffecient, use only genotype coefficient

only use the genotype FC 
```{r}
for(n in c(1:4)) {
library(dplyr)
  
genelist.up<-DEG.list[[n]] %>% dplyr::filter((UQ(as.name(names(DEG.list[[n]])[3])))>0)
genelist.down<-DEG.list[[n]] %>% dplyr::filter((UQ(as.name(names(DEG.list[[n]])[3])))<0)
    GO.ORA.temp.up<-GOseq.Atgoslim.BP.list.ORA(gsub("\\.[[:digit:]]+", "", as_vector(genelist.up[,"gene_id"])),custom.category.list=Atgoslim.TAIR.BP.list, expressed.genes = expressed.genes)
    GO.ORA.temp.down<-GOseq.Atgoslim.BP.list.ORA(gsub("\\.[[:digit:]]+", "", as_vector(genelist.down[,"gene_id"])),custom.category.list=Atgoslim.TAIR.BP.list, expressed.genes = expressed.genes)
    # handling "no enriched GO" 
    # genelist.names<-c("GO.ORA.temp.up_down","GO.ORA.temp.down_up") # test
    x<-list(GO.ORA.temp.up=GO.ORA.temp.up,
            GO.ORA.temp.down=GO.ORA.temp.down) # list
    print(x[x=="no enriched GO"])
    x<-x[!x=="no enriched GO"] # reove "no enriched GO" result
    if (length(x)!=0) {
    ## add sample info and FC info and save GO.ORA result
  for (i in 1:length(x)) {
      GO.ORA.result<-x[[i]] %>% mutate(FC = gsub("(GO.ORA.temp.)(.)","\\2",names(x)[i]),sample=DEG.objs2[n])
    save(GO.ORA.result,file=file.path("..","output",paste(gsub(".csv","",DEG.objs2[n]),gsub("(GO.ORA.temp.)(.)","\\2",names(x)[i]),"single.coef.enrich.Rdata",sep=".")))
    rm(GO.ORA.result)
  }
    }
}
```



```{r}
eGOseqs<-list.files(pattern="single.coef.enrich.Rdata",path=file.path("..", "output"))[c(1:6)]
eGOseqs.list2<-sapply(file.path("..","output",eGOseqs),function(x) mget(load(x))) # mget will return the value of the object(or objects) in a list. see https://stackoverflow.com/questions/29398630/load-data-frames-into-list
#names(eGOseqs.list2)
eGOseqs.list2.summary<-do.call("rbind",eGOseqs.list2) 
#head(eGOseqs.list2.summary) # make sure those are file names
rownames(eGOseqs.list2.summary)<-1:nrow(eGOseqs.list2.summary)
```

```{r}
library(scales)
```
#plot
```{r}
# eGOseqs.list2.summary (more than three genes in each KO)
eGOseqs.list2.summary<-eGOseqs.list2.summary[eGOseqs.list2.summary$numDEInCat>3,]
# focused on only very significant GO terms
eGOseqs.list2.summary<-eGOseqs.list2.summary[eGOseqs.list2.summary$over_represented_padjust<0.05,]
# how to cluster GO Ontology according to this pattern?
## using hclust to sort GO terms
GO.list<-unique(eGOseqs.list2.summary$category)
sample.list<-unique(eGOseqs.list2.summary$sample_FC) # revised
# having x-label also used the x-label for df below
eGOseqs.list2.summary <- eGOseqs.list2.summary %>% mutate(sample=gsub(".csv", "", sample)) %>% mutate(sample=gsub(".DEGs.int", "", sample)) %>% unite(sample_FC,sample,FC)

# Making matrix for calculate hierarchical clustering of "over-representd_padjust" value using sperad
df<-eGOseqs.list2.summary %>% dplyr::select(category,sample_FC,over_represented_padjust) %>% spread(sample_FC,over_represented_padjust,-1)

df[df<1e-100]<-1e-100 # to avoid "Inf" after log10 transformation
df<- df %>% mutate_at(vars(2:6) ,.funs=function(x) -log10(x))
# df<- df %>% mutate_at(vars(2:11) ,.funs=-log10) # does not work
df[is.na(df)]<-1
df2<-df[,2:6]
rownames(df2)<-df$category
hc<-stats::hclust(dist(df2), "ave") # only numbers
hc$order
```
#plot
```{r}
hc.tib<-tibble(category=hc$labels,order=hc$order)
# change term order using hclust() results
hc.tib2<-hc.tib %>% inner_join(eGOseqs.list2.summary[,c("category","term")],by="category") %>% distinct() %>% arrange(desc(order))
eGOseqs.list2.summary$term<-factor(eGOseqs.list2.summary$term,levels=as_vector(hc.tib2$term))
# To avoid gray color, add 1e-20 to smaller padjust value
eGOseqs.list2.summary$over_represented_padjust[eGOseqs.list2.summary$over_represented_padjust<1e-50]<-1e-50
# format x label
#eGOseqs.list2.summary <- eGOseqs.list2.summary  %>% mutate(sample2=str_replace(sample,"trt.density","trt_density")) %>%  separate(sample2,into=c("tissue","factor","DEGs","model","ref_density","ref_soil"),sep="\\.") %>% mutate(ref_soil=str_remove(ref_soil,"r")) %>% mutate(ref_density=str_remove(ref_density,"r"))  %>% unite(density_FC,ref_density,FC) # simplify elements in "density_FC" column for simpler x label in heatmap
# plot
eGOseqs.list2.summary <- separate(eGOseqs.list2.summary, "sample_FC", into = c("sample", "FC"), sep = "_")

eGOseqs.list2.summary$sample <- gsub("DEgenes\\.", "", eGOseqs.list2.summary$sample)

eGOseqs.list2.summary <- separate(eGOseqs.list2.summary, "sample", into = c("genotype", "time_point", "ref"))
  GOseq.plot<-ggplot(eGOseqs.list2.summary,aes(x=time_point,y=term)) + facet_wrap(genotype~FC, ncol = 4)+ geom_tile(aes(fill=-log10(over_represented_padjust)),colour="black")  + scale_fill_gradient2(limit=c(0,50),low=muted("green"), high=muted("magenta")) # OK
GOseq.plot<-GOseq.plot+     theme(axis.text.x=element_text(size=10,angle=90),
          axis.text.y=element_text(size=10),
          axis.title=element_text(size=10),
          axis.ticks = element_blank(),
          panel.background = element_rect(fill = "white",colour="black"),
          plot.title=element_text(size=20),
          axis.line=element_blank()) + labs(x="",y="",title="",fill="-log10\n pvalue")
GOseq.plot

ggsave(filename = "GO.enrich.heatmap.single.coef.myc102218.pdf", plot = GOseq.plot, path = "../output", height = 10, width = 10,units = "in")
```
#find out genes in each GO

```{r}
genes.in.enriched.category<-function(enrich.result,gene.list,category.table=Atgoslim.TAIR.BP.list) { # return data frame with all input data with ... which format is good? VennDiagram input format?  
  enrich.category<-enrich.result[enrich.result$over_represented_padjust<0.05,"category"]
  temp<-category.table[gene.list] # short category.table for enriched.result # gene.list shas to be character (not factor)  
  #names(temp[!is.na(names(temp))])
  temp<-temp[!is.na(names(temp))] # remove non matched AGI
  test<-data.frame(AGI=names(temp))
  for(i in enrich.category) {
    x<-rep(0,length(names(temp)))
    x[grep(i,temp)]<-1
    test<-cbind(test,x)
    names(test)[dim(test)[2]]<-i
  }
  return(test) 
}
```

```{r}
names(eGOseqs.list2) <- gsub("../output/DEgenes.", "", names(eGOseqs.list2))
names(eGOseqs.list2) <- gsub("\\.single.+", "", names(eGOseqs.list2))
names(eGOseqs.list2)
```

```{r}
names(DEG.list)
```


```{r}

names(eGOseqs.list2)[1]
names(DEG.list)[1]
#up

results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[1]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[1]] %>% filter(DEG.list[[1]][3]>0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[1], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```

```{r}

names(eGOseqs.list2)[2]
names(DEG.list)[2]
#up
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[2]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[2]] %>% filter(DEG.list[[2]][3]>0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[2], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```

```{r}

names(eGOseqs.list2)[3]
names(DEG.list)[3]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[3]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[3]] %>% filter(DEG.list[[3]][3]<0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[3], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```

```{r}

names(eGOseqs.list2)[4]
names(DEG.list)[3]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[4]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[3]] %>% filter(DEG.list[[3]][3]>0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[4], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```

```{r}

names(eGOseqs.list2)[5]
names(DEG.list)[4]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[5]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[4]] %>% filter(DEG.list[[4]][3]<0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[5], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```

```{r}

names(eGOseqs.list2)[6]
names(DEG.list)[4]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[6]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[4]] %>% filter(DEG.list[[4]][3]>0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[6], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```

```{r}
genes.in.BP.category <- list.files(path=file.path("..", "output"), pattern="GO:(\\d)+\\.enriched.csv")
genes.in.BP.category.list <- lapply(genes.in.BP.category,function(x) read_csv(paste(file.path("../output"), "/", x,sep = "")))
names(genes.in.BP.category.list) <- gsub("\\.csv", "", genes.in.BP.category)


```



```{r}
 for (i in 1:length(genes.in.BP.category.list)) {
   genes.in.BP.category.list[[i]]$category <- gsub("\\.enriched", "", names(genes.in.BP.category.list)[i])
   if (i<=9){
      genes.in.BP.category.list[[i]]<- genes.in.BP.category.list[[i]] %>% separate(category, into = c("gt","tp","ref", "up_or_down", "category"), sep = "\\.") %>% unite(sample, 3:6)
   }
   else{
     genes.in.BP.category.list[[i]]<- genes.in.BP.category.list[[i]] %>% separate(category, into = c("gt","tp","ref1", "ref2", "up_or_down", "category"), sep = "\\.") %>% unite(sample, 3:7)
   }
   
 }
```

```{r}
names(genes.in.BP.category.list)
genes.in.BP.category.list[[20]]
```


```{r}
genes.in.each.BP.category <- do.call("rbind", genes.in.BP.category.list)
genes.in.each.BP.category <- genes.in.each.BP.category[,-1]
```
add description
```{r}
load("../output/gene.description.Rdata")

genes.in.each.BP.category <- genes.in.each.BP.category %>% left_join(gene.description, by = c("AGI"="gene_id") )

##add GO terms
genes.in.each.BP.category$term <- Term(GOTERM[genes.in.each.BP.category$category])
genes.in.each.BP.category$definition <- Definition(GOTERM[genes.in.each.BP.category$category])


write.csv(genes.in.each.BP.category, file = file.path("../output", "genes.in.enriched.BP.category.csv"))
```



# use hormone responsive genes for goseq analysis



```{r}
 # bias.data vector must have the same length as DEgenes vector! 
## if you want to test expressed genes as background, add background in this function
GOseq.At.customcat.ORA<-function(genelist,padjust=0.05,custom.category.list=At.hormone.responsive.list,cdna=At_cdna,background="") { # return GO enrichment table, padjus, padjust=0.05. New BLAST2GO results with only BP Brgo.v2.5.BP.list is used (042518). Brgo.v2.5.BP.list is based on Hajar's Blast2go, which should be replaced with Brgo.Atgoslim.BP.list and giving a new name to this function (080818). cdna is DNAstring object. background is a vector of genes.
  #bias<-nchar(cdna)
  bias<-Biostrings::width(cdna)
  # add name to "bias". cdna
  #names(bias)<-names(cdna)
 names(bias) <- tibble(name=names(cdna)) %>% separate(name,into=c("name2","Symbol","description"),sep=" \\|",extra="drop",fill="left") %>%  mutate(AGI=gsub(pattern="\\.[[:digit:]]+", "", name2)) %>% pull(AGI)
 if(background=="") {print("Use all genes in genome as background")} else {
 # select only expressed geens 
  bias <- bias[background$gene_id]
  print("Expessed genes are used as background.")
  print(paste("length of bias is ",length(bias)))
 }
  TF<-(names(bias) %in% genelist)*1
  names(TF)<-names(bias)
  #print(TF)
  pwf<-nullp(TF,bias.data=bias)
  #print(pwf$DEgenes)
  GO.pval <- goseq(pwf,gene2cat=custom.category.list,use_genes_without_cat=TRUE) # format became different in new goseq version (021111). Does not work (042716)
  # calculate p ajust by BH
  GO.pval$over_represented_padjust<-p.adjust(GO.pval$over_represented_pvalue,method="BH")
  if(GO.pval$over_represented_padjust[1]>padjust) return("no enriched GO")
  else {
    enriched.GO<-GO.pval[GO.pval$over_represented_padjust<padjust,] 
    print("enriched.GO is")
    print(enriched.GO)
    return(enriched.GO)
  }
}
```


```{r}
for(n in c(1,2,3,5,7,9,11,13)) {
library(dplyr)
  
genelist.up<-DEG.list[[n]] %>% dplyr::filter((UQ(as.name(names(DEG.list[[n]])[3])))>0)
genelist.down<-DEG.list[[n]] %>% dplyr::filter((UQ(as.name(names(DEG.list[[n]])[3])))<0)
    GO.ORA.temp.up<-GOseq.At.customcat.ORA(gsub("\\.[[:digit:]]+", "", as_vector(genelist.up[,"gene_id"])), background = expressed.genes)
    GO.ORA.temp.down<-GOseq.At.customcat.ORA(gsub("\\.[[:digit:]]+", "", as_vector(genelist.down[,"gene_id"])), background = expressed.genes)
    # handling "no enriched GO" 
    # genelist.names<-c("GO.ORA.temp.up_down","GO.ORA.temp.down_up") # test
    x<-list(GO.ORA.temp.up=GO.ORA.temp.up,
            GO.ORA.temp.down=GO.ORA.temp.down) # list
    print(x[x=="no enriched GO"])
    x<-x[!x=="no enriched GO"] # reove "no enriched GO" result
    if (length(x)!=0) {
    ## add sample info and FC info and save GO.ORA result
  for (i in 1:length(x)) {
      GO.ORA.result<-x[[i]] %>% mutate(FC = gsub("(GO.ORA.temp.)(.)","\\2",names(x)[i]),sample=DEG.objs2[n])
    save(GO.ORA.result,file=file.path("..","output",paste(gsub(".csv","",DEG.objs2[n]),gsub("(GO.ORA.temp.)(.)","\\2",names(x)[i]),"single.coef.enrich.hormone.responsive.Rdata",sep=".")))
    rm(GO.ORA.result)
  }
    }
}
```

```{r}


eGOseqs<-list.files(pattern="single.coef.enrich.hormone.responsive.Rdata",path=file.path("..", "output"))[1:7]
eGOseqs.list2<-sapply(file.path("..","output",eGOseqs),function(x) mget(load(x))) # mget will return the value of the object(or objects) in a list. see https://stackoverflow.com/questions/29398630/load-data-frames-into-list
#names(eGOseqs.list2)
eGOseqs.list2.summary<-do.call("rbind",eGOseqs.list2) 
#head(eGOseqs.list2.summary) # make sure those are file names
rownames(eGOseqs.list2.summary)<-1:nrow(eGOseqs.list2.summary)


```

```{r}
# eGOseqs.list2.summary (more than three genes in each KO)
eGOseqs.list2.summary<-eGOseqs.list2.summary[eGOseqs.list2.summary$numDEInCat>3,] # I see
# focused on only very significant GO terms
eGOseqs.list2.summary<-eGOseqs.list2.summary[eGOseqs.list2.summary$over_represented_padjust<0.05,]
# how to cluster GO Ontology according to this pattern?
## using hclust to sort GO terms
GO.list<-unique(eGOseqs.list2.summary$category)
sample.list<-unique(eGOseqs.list2.summary$sample_FC) # revised
# having x-label also used the x-label for df below
eGOseqs.list2.summary <- eGOseqs.list2.summary %>% mutate(sample=gsub(".csv", "", sample)) %>% mutate(sample=gsub(".DEGs.int", "", sample)) %>% unite(sample_FC,sample,FC)

# Making matrix for calculate hierarchical clustering of "over-representd_padjust" value using sperad
df<-eGOseqs.list2.summary %>% dplyr::select(category,sample_FC,over_represented_padjust) %>% spread(sample_FC,over_represented_padjust,-1)

df[df<1e-100]<-1e-100 # to avoid "Inf" after log10 transformation
df<- df %>% mutate_at(vars(2:8) ,.funs=function(x) -log10(x))
# df<- df %>% mutate_at(vars(2:11) ,.funs=-log10) # does not work
df[is.na(df)]<-1
df2<-df[,2:8]
rownames(df2)<-df$category
hc<-stats::hclust(dist(df2), "ave") # only numbers
hc$order
```

```{r}
hc.tib<-tibble(category=hc$labels,order=hc$order)
# change term order using hclust() results
hc.tib2<-hc.tib %>% arrange(desc(order))
#eGOseqs.list2.summary$term<-factor(eGOseqs.list2.summary$term,levels=as_vector(hc.tib2$term))
# To avoid gray color, add 1e-20 to smaller padjust value
eGOseqs.list2.summary$over_represented_padjust[eGOseqs.list2.summary$over_represented_padjust<1e-50]<-1e-50
# format x label
#eGOseqs.list2.summary <- eGOseqs.list2.summary  %>% mutate(sample2=str_replace(sample,"trt.density","trt_density")) %>%  separate(sample2,into=c("tissue","factor","DEGs","model","ref_density","ref_soil"),sep="\\.") %>% mutate(ref_soil=str_remove(ref_soil,"r")) %>% mutate(ref_density=str_remove(ref_density,"r"))  %>% unite(density_FC,ref_density,FC) # simplify elements in "density_FC" column for simpler x label in heatmap
# plot
eGOseqs.list2.summary <- separate(eGOseqs.list2.summary, "sample_FC", into = c("sample", "FC"), sep = "_")

eGOseqs.list2.summary$sample <- gsub("DEgenes\\.", "", eGOseqs.list2.summary$sample)

eGOseqs.list2.summary <- separate(eGOseqs.list2.summary, "sample", into = c("genotype", "time_point", "ref"))
  #GOseq.plot<-ggplot(eGOseqs.list2.summary,aes(x=time_point,y=term)) + facet_wrap(genotype~FC, ncol = 4)+ geom_tile(aes(fill=-log10(over_represented_padjust)),colour="black")  + scale_fill_gradient2(limit=c(0,50),low=muted("green"), high=muted("magenta")) # OK
#GOseq.plot<-GOseq.plot+     theme(axis.text.x=element_text(size=10,angle=90),
          #axis.text.y=element_text(size=10),
          #axis.title=element_text(size=10),
          #axis.ticks = element_blank(),
          #panel.background = element_rect(fill = "white",colour="black"),
          #plot.title=element_text(size=20),
          #axis.line=element_blank()) + labs(x="",y="",title="",fill="-log10\n pvalue")
#GOseq.plot

  GOseq.plot<-ggplot(eGOseqs.list2.summary,aes(x=time_point,y=category)) + facet_wrap(genotype~FC, ncol = 4)+ geom_tile(aes(fill=-log10(over_represented_padjust)),colour="black")  + scale_fill_gradient2(limit=c(0,50),low=muted("green"), high=muted("magenta")) # OK
GOseq.plot<-GOseq.plot+     theme(axis.text.x=element_text(size=10,angle=90),
          axis.text.y=element_text(size=10),
          axis.title=element_text(size=10),
          axis.ticks = element_blank(),
          panel.background = element_rect(fill = "white",colour="black"),
          plot.title=element_text(size=20),
          axis.line=element_blank()) + labs(x="",y="",title="",fill="-log10\n pvalue")
GOseq.plot

ggsave(filename = "GO.enrich.heatmap.single.coef.hormone.responsive.102218.pdf", plot = GOseq.plot, path = "../output", height = 5, width = 5,units = "in")
```


```{r}
DEG.list
```

```{r}
genes.in.enriched.category<-function(enrich.result,gene.list,category.table=At.hormone.responsive.list) { # return data frame with all input data with ... which format is good? VennDiagram input format?  
  enrich.category<-enrich.result[enrich.result$over_represented_padjust<0.05,"category"]
  temp<-category.table[gene.list] # short category.table for enriched.result # gene.list shas to be character (not factor)  
  #names(temp[!is.na(names(temp))])
  temp<-temp[!is.na(names(temp))] # remove non matched AGI
  test<-data.frame(AGI=names(temp))
  for(i in enrich.category) {
    x<-rep(0,length(names(temp)))
    x[grep(i,temp)]<-1
    test<-cbind(test,x)
    names(test)[dim(test)[2]]<-i
  }
  return(test) 
}
```

```{r}
names(eGOseqs.list2)
names(eGOseqs.list2) <- gsub("../output/DEgenes.", "", names(eGOseqs.list2))
names(eGOseqs.list2) <- gsub(".single.+", "", names(eGOseqs.list2))
```
```{r}
names(eGOseqs.list2)
names(DEG.list)
```


```{r}
names(eGOseqs.list2)[1]
names(DEG.list)[1]
#down
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[1]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[1]] %>% filter(DEG.list[[1]][3]<0))$gene_id))
ncol(results.table)
results.table.Col.1h.rH.down.hormone.responsive <- results.table %>% filter(results.table[,2]==1) %>% dplyr::select(AGI)
write.csv(results.table.Col.1h.rH.down.hormone.responsive, file = file.path("../output", paste(names(eGOseqs.list2)[1], colnames(results.table)[2],"enriched.csv", sep = ".")))


```

```{r}
names(eGOseqs.list2)[2]
names(DEG.list)[1]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[2]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[1]] %>% filter(DEG.list[[1]][3]>0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[2], colnames(results.table)[i],"enriched.csv", sep = ".")))
}

```


```{r}
names(eGOseqs.list2)[3]
names(DEG.list)[2]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[3]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[2]] %>% filter(DEG.list[[2]][3]>0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[3], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```

```{r}
names(eGOseqs.list2)[4]
names(DEG.list)[3]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[4]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[3]] %>% filter(DEG.list[[3]][3]<0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[4], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```

```{r}
names(eGOseqs.list2)[5]
names(DEG.list)[3]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[5]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[3]] %>% filter(DEG.list[[3]][3]>0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[5], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```

```{r}
names(eGOseqs.list2)[6]
names(DEG.list)[4]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[6]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[4]] %>% filter(DEG.list[[4]][3]<0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[6], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```


```{r}
names(eGOseqs.list2)[7]
names(DEG.list)[4]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[7]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",(DEG.list[[4]] %>% filter(DEG.list[[4]][3]>0))$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[7], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```
#below is for npr1, 
```{r}
names(eGOseqs.list2)[8]
names(DEG.list)[7]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[8]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",DEG.list[[7]]$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[8], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```


```{r}
names(eGOseqs.list2)[9]
names(DEG.list)[9]
results.table<-genes.in.enriched.category(enrich.result=eGOseqs.list2[[9]],gene.list=gsub("(.)(\\.[[:digit:]]+)","\\1",DEG.list[[9]]$gene_id))

for(i in 2:ncol(results.table)) {
  temp <- results.table %>% filter(results.table[,i]==1) %>% dplyr::select(AGI)
write.csv(temp, file = file.path("../output", paste(names(eGOseqs.list2)[9], colnames(results.table)[i],"enriched.csv", sep = ".")))
}
```
# find out genes in each hormone responsive category
```{r}
genes.in.hormone.responsive.category <- list.files(path=file.path("..", "output"), pattern="enriched.csv")
genes.in.hormone.responsive.category <- genes.in.hormone.responsive.category[-grep("GO:(\\d)+", genes.in.hormone.responsive.category)]
genes.in.hormone.responsive.category.list <- lapply(genes.in.hormone.responsive.category,function(x) read_csv(paste(file.path("../output"), "/", x,sep = "")))
names(genes.in.hormone.responsive.category.list) <- gsub("\\.csv", "", genes.in.hormone.responsive.category)


```

```{r}
#head(Description.merge)
#head (gene.description)
#gene.description <- gene.description %>% left_join(Description.merge, by=c("gene_id"="ID"))
#save(gene.description, file = file.path("../output", "gene.description.Rdata"))
```

```{r}
 for (i in 1:length(genes.in.hormone.responsive.category.list)) {
   genes.in.hormone.responsive.category.list[[i]]$category <- gsub("\\.enriched", "", names(genes.in.hormone.responsive.category.list)[i])
   if (i<=7){
      genes.in.hormone.responsive.category.list[[i]]<- genes.in.hormone.responsive.category.list[[i]] %>% separate(category, into = c("gt","tp","ref", "up_or_down", "category"), sep = "\\.") %>% unite(sample, 3:6)
   }
   else{
     genes.in.hormone.responsive.category.list[[i]]<- genes.in.hormone.responsive.category.list[[i]] %>% separate(category, into = c("gt","tp","ref1", "ref2", "up_or_down", "category"), sep = "\\.") %>% unite(sample, 3:7)
   }
   
 }
```

```{r}
genes.in.each.hormone.responsive.category <- do.call("rbind", genes.in.hormone.responsive.category.list)
genes.in.each.hormone.responsive.category <- genes.in.each.hormone.responsive.category[,-1]
```
add description
```{r}
genes.in.each.hormone.responsive.category <- genes.in.each.hormone.responsive.category %>% left_join(gene.description, by = c("AGI"="gene_id") )

write.csv(genes.in.each.hormone.responsive.category, file = file.path("../output", "genes.in.enriched.hormone.responsive.category.csv"))
```

