# SAS_defense_transcriptome
## Please download data from following places:
## For mapping reads
* raw fastq files: chunmei@whitney:~/RNAseq/trimmed
* Arabidopsis thaliana cDNA reference sequences: chunmei@whitney:~/RNAseq/reference/TAIR10_cdna_20110103_representative_gene_model_updated

## For differential gene expression analysis
* Saved edgeR dge object ("dge.Rdata"): ~/Box/Chunmei_Myc_Paper/manuscripts/output/dge.Rdata
* Arabidopsis thaliana gene annotationta: chunmei@whitney:~/RNAseq/output/gene.description.Rdata

## explanation of DEG file names
Please find DEG files under DEGs_with_descritipn folder (KN; where the description has been added in the script?).
Or should I use DEGs files under "figure_and_tables"? If so, we need to replace "DEGs_with_description" into "figure_and_tables" in all scripts used.

DEG files in "~/Box/Chunmei_Myc_Paper/manuscripts/output/DEGs_with_description", "~/Box/Chunmei_Myc_Paper/manuscripts/output/figures_and_tables", and "~/Box/Chunmei_Myc_Paper/manuscripts/Figures and tables" are the same, I added the detailed description of each DEG files at the end of the manuscipt. Because I might move the DEG files generated with out saving the scipts to other folders and also might rename the file not through R, I don't think there is description in the script.

RNAseq data was splitted by time points (1h or 49h).

* Interaction model (gt*trt) RNAseq data analysis.Rmd
    + DEgenes.Col.1h.rH.csv: Col, 1h, shade responsive genes (reference is H).
    + DEgenes.Col.49h.4H.csv: Col, 49h, shade responsive genes (reference is H)
    + DEgenes.myc234.1h.rCol.rH.csv: myc234, 1h, (double coefficients, c("gtmyc234", "gtmyc234:trtL") )

* subsetted by genotype further?
    + DEgenes.myc234.1h.rH.csv: myc234, 1h, (KN, what is the difference between "DEgenes.myc234.1h.rCol.rH.csv" and "DEgenes.myc234.1h.rH.csv"? )

"DEgenes.myc234.1h.rCol.rH.csv" was generated by using Col or Col under H as reference, "DEgenes.myc234.1h.rH.csv" was generated by using the genotype itself myc234 under H as reference.


## GO term over-representation analysis (GOseq_analysis_CL.Rmd)
* Arabidopsis thaliana GOslim 

## GO term heatmap (GOseq_heatmap2.Rmd)

## gene clustering (DE_genes_heatmap_scale_center_f.Rmd)

## Schweizer (2013) myc234 transcriptome data reanalized (2013_data_reanalized.Rmd)

## Detailed analysis of PIFtargets and myc234 misregulated genes (MYC234_PIF4target.Rmd, .md, .html, MYC234_PIF4target_files) (by Kazu)

## Differentially expressed genes overlap between different conditions (101218differentially expressed genes.Rmd)

## All output files are stoerd in output folder not included in this repository.
  |-input
  
  |-output
  
  |-SAS_defense_transcriptome (this repository)
 
 



