---
title: "SAS defense transcriptome"
output: html_document
---

```{r}
#test regular expression
stringtest <- "CL_sid_49h_H_R3_S189_trimmed.fq.gz_output"
gsub("_S\\d\\d\\d_trimmed.fq.gz_output", "", stringtest)
#test to see how abundance.tsv look like
test1 <- read.table("~/RNAseq/map_output2/CL_sid_49h_H_R3_S189_trimmed.fq.gz_output/abundance.tsv", header = T)
head(test1)
test1 $ sample_id <- "CL_sid_49h_H_R3"
head(test1)
```

add sample_id to each tsv file
```{r}
common_path <- "~/RNAseq/map_output2/"
dir_list <-list.files("~/RNAseq/map_output2/") 
dir_list

total_abundance <- data.frame(matrix(ncol = ncol(test1), nrow=0))
colnames(total_abundance) <- colnames(test1)
for (i in dir_list){
  temp_data <- read.table(file = paste(common_path, i, "/abundance.tsv", sep = ""), header = T)
 
  temp_data$sample_id <- gsub("_S\\d\\d\\d_trimmed.fq.gz_output", "", i) 
   #add sample_id for each data.frame 
  
 
  total_abundance <- rbind(total_abundance, temp_data)
  
}

head(total_abundance)
total_abundance$sample_id <- gsub("CL_", "", total_abundance$sample_id)
head(total_abundance)
```

plot counts for each sample

```{r}
library(tidyverse)
#keep only target_id, est_counts, and sample_id
counts.data <- dplyr::select(total_abundance, c(1,4,6))
colnames(counts.data)[2] <- "counts"
head(counts.data)
summary(counts.data)
#plot histgram

#write a function below
#counts.Col <- counts.data[grep("Col", counts.data$sample_id), ]
#counts.transformed.hist.Col <- ggplot(counts.Col)+geom_histogram(aes(x=log10(counts)))+ facet_wrap(~sample_id, ncol = 4)
#counts.transformed.hist.Col

#counts.myc234 <- counts.data[grep("myc234", counts.data$sample_id), ]
#counts.transformed.hist.myc234 <- ggplot(counts.myc234)+geom_histogram(aes(x=log10(counts)))+ facet_wrap(~sample_id, ncol = 4)
#counts.transformed.hist.myc234

hist.genotype <- function(data, genotype){
  counts.genotype <- counts.data[grep(genotype, data$sample_id), ]
  counts.transformed.hist.genotype <- ggplot(counts.genotype)+geom_histogram(aes(x=log10(counts)))+ facet_wrap(~sample_id, ncol = 4)
counts.transformed.hist.genotype
}

hist.genotype(counts.data, "npr")
```


spread counts.data for pairs

```{r}
counts.data.spread <- counts.data %>% 
  spread(key=sample_id, value= counts)

head(counts.data.spread)

# retain the genes with >10 reads in 3 samples
# currently don't quite understand
counts.data.spread <- counts.data.spread[rowSums(counts.data.spread[,-1] > 10) >= 3,]

head(counts.data.spread)
```

test for first 1000 genes

```{r}
counts.data.1000 <- head(counts.data.spread[,-1],n=1000)
head(counts.data.1000)
#image is too large to plot 64 samples, try 1000genes in Col first 
p <- pairs(counts.data.1000[ ,grep("Col", colnames(counts.data.1000))], log="xy")
p
ggsave("plot_pair_sid_full.png",path=paste(getwd(),"/figures", sep = ""),pairs(counts.data.spread[ ,grep("sid", colnames(counts.data.spread))], log="xy"),width=20, height=20, units = "cm")

cor(counts.data.spread[,6:9])
```

assign groups

```{r}
sample.description <- tibble(sample= colnames(counts.data.spread)[-1])
head(sample.description)
library(stringr)
sample.description <- sample.description %>% 
  #This next line searches for genotypes in the colnames of counts.data.spread and returns anything that matches
  #In this way we can extract the genotype info.
  mutate(gt=str_extract(sample,"Col|myc234|npr|sid")) %>%
  
  #Now we use the same method to get the treatment
  mutate(trt=str_extract(sample,"H|L")) %>%
  #Now for time points
  mutate(tp=str_extract(sample, "1h|49h")) %>%
  
  # paste the trt, gt, tp columns together to give a group identifier

  mutate(group=str_c(gt,trt,tp, sep = "_"))

sample.description

#next convert gt and trt into group variables (factors)

sample.description <- sample.description %>%
  mutate(gt=factor(gt, levels = c("Col", "myc234", "npr", "sid")), 
         trt=factor(trt,levels = c("H","L")),
         tp=factor(tp, levels = c("1h", "49h"))
         ) # setting the levels in this way makes "H" the refernce  

sample.description
```

calculate normalization factor

```{r}
library(edgeR)
counts.matrix <- counts.data.spread %>% dplyr::select(-target_id) %>% as.matrix()
rownames(counts.matrix) <- counts.data.spread$target_id
#colnames(counts.matrix) <- gsub("CL_", "", colnames(counts.matrix))
dge.data <- DGEList(counts=counts.matrix, 
                    samples = sample.description,
                    genes = counts.data.spread$target_id,
                    group=sample.description$group)
dim(dge.data) 
dge.data <- calcNormFactors(dge.data, method = "TMM")
dge.data$samples # look at the normalization factors
```
make normalization factor plot

```{r}

ggplot(dge.data$samples,aes(x=sample,y=norm.factors,fill=trt)) + geom_col() + 
  theme(axis.text.x  = element_text(angle=90, vjust=0.5,size = 7))
```
Col MDS plot only

```{r}
#subset genotype
dge.Col <- dge.data[,dge.data$samples$gt=="Col"]
mds.Col <- plotMDS(dge.Col,method = "bcv",labels=dge.Col$samples$group,gene.selection = "pairwise",dim.plot = c(1,2),ndim=3,plot=FALSE)
mds.Col.pl <- as_tibble(mds.Col$cmdscale.out) %>% 
  bind_cols(data.frame(sample=row.names(mds.Col$cmdscale.out)),.) %>% 
  inner_join(dge.data$samples)

mds.Col.pl %>% ggplot(aes(x=V1,y=V2, shape=tp, color=trt)) + geom_point(size=3) + ggtitle("Col Samples DIM 1 vs 2")+facet_wrap(~tp)
```


myc234 MDS plot only

```{r}
#subset genotype
dge.myc234 <- dge.data[,dge.data$samples$gt=="myc234"]
mds.myc234 <- plotMDS(dge.myc234,method = "bcv",labels=dge.myc234$samples$group,gene.selection = "pairwise",dim.plot = c(1,2),ndim=3,plot=FALSE)
mds.myc234.pl <- as_tibble(mds.myc234$cmdscale.out) %>% 
  bind_cols(data.frame(sample=row.names(mds.myc234$cmdscale.out)),.) %>% 
  inner_join(dge.data$samples)

mds.myc234.pl %>% ggplot(aes(x=V1,y=V2, shape=tp, color=trt)) + geom_point(size=3) + ggtitle("myc234 Samples DIM 1 vs 2")+facet_wrap(~tp)
```

npr MDS plot only

```{r}
#subset genotype
dge.npr <- dge.data[,dge.data$samples$gt=="npr"]
mds.npr <- plotMDS(dge.npr,method = "bcv",labels=dge.npr$samples$group,gene.selection = "pairwise",dim.plot = c(1,2),ndim=3,plot=FALSE)
mds.npr.pl <- as_tibble(mds.npr$cmdscale.out) %>% 
  bind_cols(data.frame(sample=row.names(mds.npr$cmdscale.out)),.) %>% 
  inner_join(dge.data$samples)

mds.npr.pl %>% ggplot(aes(x=V1,y=V2, shape=tp, color=trt)) + geom_point(size=3) + ggtitle("npr Samples DIM 1 vs 2")+facet_wrap(~tp)
```


sid MDS plot only

```{r}
#subset genotype
dge.sid <- dge.data[,dge.data$samples$gt=="sid"]
mds.sid <- plotMDS(dge.sid,method = "bcv",labels=dge.sid$samples$group,gene.selection = "pairwise",dim.plot = c(1,2),ndim=3,plot=FALSE)
mds.sid.pl <- as_tibble(mds.sid$cmdscale.out) %>% 
  bind_cols(data.frame(sample=row.names(mds.sid$cmdscale.out)),.) %>% 
  inner_join(dge.data$samples)

mds.sid.pl %>% ggplot(aes(x=V1,y=V2, shape=tp, color=trt)) + geom_point(size=3) + ggtitle("sid Samples DIM 1 vs 2")+facet_wrap(~tp)
```


Make a plot of the Biological Coefficient of Variation of each sample
```{r}
plotMDS(dge.data, method = "bcv")
```
Do MDS for all data and facet by gt~tp
```{r}
mds.all <- plotMDS(dge.data, method = "bcv", labels=dge.data$samples$group,gene.selection = "pairwise",dim.plot = c(1,2),ndim=3,plot=FALSE)
mds.all$cmdscale.out

mds.all.pl <- as_tibble(mds.all$cmdscale.out) %>% 
  bind_cols(data.frame(sample=row.names(mds.all$cmdscale.out)),.) %>% 
  inner_join(dge.data$samples)

mds.all.pl %>% ggplot(aes(x=V1,y=V2, color= trt)) + geom_point(size=3) + ggtitle("all Samples DIM 1 vs 2") + facet_grid(gt~tp)
```
extract normalized data
```{r}
counts.data.normal <- cpm(dge.data)
head(counts.data.normal, n =6)
```
before normalization
```{r}

boxplot(log2(counts.data.spread[,-1]+1))
```
boxplot for counts after normalization
```{r}
boxplot(log2(counts.data.normal+1))
```







plot genes

```{r}
library(reshape2)
plotDE <- function(genes, dge, sample.description) {
  require(ggplot2)
  require(reshape2)
  tmp.data <- t(log2(cpm(dge[genes,])+1))
  tmp.data <- merge(tmp.data,sample.description,by.x="row.names",by.y="sample")
  tmp.data <- melt(tmp.data,value.name="log2_cpm",variable.name="gene")
  pl <- ggplot(tmp.data,aes(x=gt,y=log2_cpm,fill=trt))
  pl <- pl + facet_grid(gene~ tp)
  pl <- pl + ylab("log2(cpm)") + xlab("genotype")
  pl <- pl + geom_boxplot()
  pl + theme(axis.text.x  = element_text(angle=45, vjust=1,hjust=1))
}
```




#subset Col in 1h treatment
#is it correct?? few genes try another way, subset the counts data first
```{r}
#dge.data.1h.Col <- dge.data[,grep("Col_1h", dge.data$samples$sample)]
#dge.data.1h.Col
#subset sample.description as well
#sample.description.Col.1h <- sample.description[grep("Col_1h", sample.description$sample),]
#sample.description.Col.1h <- droplevels(sample.description.Col.1h)

```

#assign groups Col 1h treatment

```{r}
#subset counts.data.spread
counts.data.spread.Col.1h <- counts.data.spread %>% dplyr::select(1:9)
sample.description.Col.1h <- tibble(sample=colnames(counts.data.spread.Col.1h)[-1])
sample.description.Col.1h
library(stringr)
sample.description.Col.1h <- sample.description.Col.1h %>% 
  #This next line searches for genotypes in the colnames of counts.data.spread and returns anything that matches
  #In this way we can extract the genotype info.
  mutate(gt=str_extract(sample,"Col")) %>%
  
  #Now we use the same method to get the treatment
  mutate(trt=str_extract(sample,"H|L")) %>%
  #Now for time points
  mutate(tp=str_extract(sample, "1h")) %>%
  
  # paste the trt, gt, tp columns together to give a group identifier

  mutate(group=str_c(gt,trt,tp, sep = "_"))

sample.description.Col.1h

#next convert gt and trt into group variables (factors)

sample.description.Col.1h <- sample.description.Col.1h %>%
  mutate(gt=as.factor(gt), 
         trt=factor(trt,levels = c("H","L")),
         tp=as.factor(tp)
         ) # setting the levels in this way makes "H" the refernce  

sample.description.Col.1h
```

calculate normalization factor

```{r}
library(edgeR)
counts.matrix.Col.1h <- counts.data.spread.Col.1h %>% dplyr::select(-target_id) %>% as.matrix()
rownames(counts.matrix.Col.1h) <- counts.data.spread.Col.1h$target_id
colnames(counts.matrix.Col.1h) <- gsub("CL_", "", colnames(counts.matrix.Col.1h))
dge.data.1h.Col <- DGEList(counts=counts.matrix.Col.1h, 
                    samples = sample.description.Col.1h,
                    genes = counts.data.spread.Col.1h$target_id,
                    group=sample.description.Col.1h$group)
dim(dge.data.1h.Col) 
dge.data.1h.Col <- calcNormFactors(dge.data.1h.Col, method = "TMM")
dge.data.1h.Col$samples # look at the normalization factors
```
#design Col_1h
```{r}
design.Col.1h <- model.matrix(~trt,data = sample.description.Col.1h)
rownames(design.Col.1h) <- sample.description.Col.1h$sample
design.Col.1h
```


```{r}
#First the overall dispersion
dge.data.1h.Col <- estimateGLMCommonDisp(dge.data.1h.Col,design.Col.1h,verbose = TRUE)

#Then a trended dispersion based on count level
dge.data.1h.Col <- estimateGLMTrendedDisp(dge.data.1h.Col,design.Col.1h)

#And lastly we calculate the gene-wise dispersion, using the prior estimates to "squeeze" the dispersion towards the common dispersion.
dge.data.1h.Col <- estimateGLMTagwiseDisp(dge.data.1h.Col,design.Col.1h)

#We can examine this with a plot
plotBCV(dge.data.1h.Col)
```



#just for the genes differentially expressed in Col under 1h shade treatment, 
```{r}
fit.Col.1h <- glmFit(dge.data.1h.Col, design.Col.1h)
Col.1h.lrt <- glmLRT(fit.Col.1h,coef = "trtL")
topTags(Col.1h.lrt) # the top 10 most differentially expressed genes
summary(decideTestsDGE(Col.1h.lrt,p.value=0.01)) #This uses the FDR.
summary(decideTestsDGE(Col.1h.lrt,p.value=0.05)) #This uses the FDR.  0.05 would be OK also.
```
#save the differentially expressed genes at 0.05

```{r}
#Extract genes with a FDR < 0.01 (could also use 0.05)
DEgene.Col.1h <- topTags(Col.1h.lrt,n = Inf,p.value = 0.05)$table
rownames(DEgene.Col.1h) <- NULL
DEgene.Col.1h <- column_to_rownames(DEgene.Col.1h, var = "genes")
#save to a file
getwd()
write.csv(DEgene.Col.1h,"../output/DEgenes.Col.1h.csv")
```

#plot genes differentially expressed at 1h shade treatment.

```{r}
plotDE(rownames(DEgene.Col.1h)[1:3],dge.data.1h.Col,sample.description.Col.1h)
```



#subset Col in 49h treatment
```{r}
#dge.data.49h.Col <- dge.data[,grep("Col_49h", dge.data$samples$sample)]
#dge.data.49h.Col
#subset sample.description as well
#sample.description.Col.49h <- sample.description[grep("Col_49h", sample.description$sample),]
#sample.description.Col.49h <- droplevels(sample.description.Col.49h)
#sample.description.Col.49h
```

#assign groups Col 49h treatment

```{r}
#subset counts.data.spread
counts.data.spread.Col.49h <- counts.data.spread %>% dplyr::select(1, 10:17)
sample.description.Col.49h <- tibble(sample=colnames(counts.data.spread.Col.49h)[-1])
sample.description.Col.49h
library(stringr)
sample.description.Col.49h <- sample.description.Col.49h %>% 
  #This next line searches for genotypes in the colnames of counts.data.spread and returns anything that matches
  #In this way we can extract the genotype info.
  mutate(gt=str_extract(sample,"Col")) %>%
  
  #Now we use the same method to get the treatment
  mutate(trt=str_extract(sample,"H|L")) %>%
  #Now for time points
  mutate(tp=str_extract(sample, "49h")) %>%
  
  # paste the trt, gt, tp columns together to give a group identifier

  mutate(group=str_c(gt,trt,tp, sep = "_"))

sample.description.Col.49h

#next convert gt and trt into group variables (factors)

sample.description.Col.49h <- sample.description.Col.49h %>%
  mutate(gt=as.factor(gt), 
         trt=factor(trt,levels = c("H","L")),
         tp=as.factor(tp)
         ) # setting the levels in this way makes "H" the refernce  

sample.description.Col.49h
```

calculate normalization factor

```{r}
library(edgeR)
counts.matrix.Col.49h <- counts.data.spread.Col.49h %>% dplyr::select(-target_id) %>% as.matrix()
rownames(counts.matrix.Col.49h) <- counts.data.spread.Col.49h$target_id
colnames(counts.matrix.Col.49h) <- gsub("CL_", "", colnames(counts.matrix.Col.49h))
dge.data.49h.Col <- DGEList(counts=counts.matrix.Col.49h, 
                    samples = sample.description.Col.49h,
                    genes = counts.data.spread.Col.49h$target_id,
                    group=sample.description.Col.49h$group)
dim(dge.data.49h.Col) 
dge.data.49h.Col <- calcNormFactors(dge.data.49h.Col, method = "TMM")
dge.data.49h.Col$samples # look at the normalization factors
```
#design Col_49h

```{r}
design.Col.49h <- model.matrix(~trt,data = sample.description.Col.49h)
rownames(design.Col.49h) <- sample.description.Col.49h$sample
design.Col.49h
```

```{r}
#First the overall dispersion
dge.data.49h.Col <- estimateGLMCommonDisp(dge.data.49h.Col,design.Col.49h,verbose = TRUE)

#Then a trended dispersion based on count level
dge.data.49h.Col <- estimateGLMTrendedDisp(dge.data.49h.Col,design.Col.49h)

#And lastly we calculate the gene-wise dispersion, using the prior estimates to "squeeze" the dispersion towards the common dispersion.
dge.data.49h.Col <- estimateGLMTagwiseDisp(dge.data.49h.Col,design.Col.49h)

#We can examine this with a plot
plotBCV(dge.data.49h.Col)
```



#just for the genes differentially expressed in Col under 49h shade treatment, 
```{r}
fit.Col.49h <- glmFit(dge.data.49h.Col, design.Col.49h)
Col.49h.lrt <- glmLRT(fit.Col.49h,coef = "trtL")
topTags(Col.49h.lrt) # the top 10 most differentially expressed genes
summary(decideTestsDGE(Col.49h.lrt,p.value=0.01)) #This uses the FDR.
summary(decideTestsDGE(Col.49h.lrt,p.value=0.05)) #This uses the FDR.  0.05 would be OK also.
```
#save the differentially expressed genes at 0.05

```{r}
#Extract genes with a FDR < 0.01 (could also use 0.05)
DEgene.Col.49h <- topTags(Col.49h.lrt,n = Inf,p.value = 0.05)$table
rownames(DEgene.Col.49h) <- NULL
DEgene.Col.49h <- column_to_rownames(DEgene.Col.49h, var = "genes")
#save to a file
getwd()
write.csv(DEgene.Col.49h,"DEgenes.Col.49h.csv")
```

#plot genes differentially expressed at 49h shade treatment.

```{r}
plotDE(rownames(DEgene.Col.49h)[1:3],dge.data.49h.Col,sample.description.Col.49h)
```

```{r}
DEgenecpm.1h <- cpm(dge.data.1h.Col[rownames(DEgene.Col.1h), ])
write.csv(DEgenecpm.1h, "DEgenecpm.1h.csv")
DEgenecpm.49h <- cpm(dge.data.49h.Col[rownames(DEgene.Col.49h), ])
write.csv(DEgenecpm.49h, "DEgenecpm.49h.csv")
```

